controller/array_action/array_mediator_svc.py

```diff
@@ -1377,3 +1402,90 @@ def _add_snapshot(self, snapshot_name, source_cli_volume, pool):
         if cli_snapshot is None:
             raise array_errors.ObjectNotFoundError(snapshot_id)
         return cli_snapshot
+
+    def _mkvolumegroup(self, name, pool, io_group, source_id):
+        cli_kwargs = {
+            'type': 'clone',
+            'fromsnapshotid': source_id,
+            'pool': pool,
+            'name': name
+        }
+        if io_group:
+            cli_kwargs['iogroup'] = io_group
+        try:
+            return self.client.svctask.mkvolumegroup(**cli_kwargs)
+        except (svc_errors.CommandExecutionError, CLIFailureError) as ex:
+            if is_warning_message(ex.my_message):
+                logger.warning("exception encountered during creation of volume {0}: {1}".format(name, ex.my_message))
+            else:
+                logger.error("Cannot create volume {0}, Reason is: {1}".format(name, ex.my_message))
+                if OBJ_ALREADY_EXIST in ex.my_message:
+                    raise array_errors.VolumeAlreadyExists(name, self.endpoint)
+                if NAME_NOT_EXIST_OR_MEET_RULES in ex.my_message:
+                    raise array_errors.PoolDoesNotExist(pool, self.endpoint)
+                if NOT_ENOUGH_EXTENTS_IN_POOL_CREATE in ex.my_message:
+                    raise array_errors.NotEnoughSpaceInPool(id_or_name=pool)
+                if any(msg_id in ex.my_message for msg_id in (NON_ASCII_CHARS, INVALID_NAME, TOO_MANY_CHARS)):
+                    raise array_errors.InvalidArgumentError(ex.my_message)
+                raise ex
+        return None
+
+    def _get_volume_id_from_volume_group(self, volume_group_id):
+        filter_value = 'volume_group_id={}'.format(volume_group_id)
+        cli_volume = self._lsvdisk(filtervalue=filter_value)
+        return cli_volume.id
+
+    def _rollback_create_volume_from_snapshot(self, volume_id, volume_group_id):
+        self._delete_volume_by_name(volume_id)
+        self._rmvolumegroup(volume_group_id)
+
+    def _change_volume_group(self, volume_id, volume_group):
+        cli_kwargs = {}
+        if volume_group:
+            cli_kwargs['volumegroup'] = volume_group
+        else:
+            cli_kwargs['novolumegroup'] = True
+        self._chvdisk(volume_id, **cli_kwargs)
+
+    def _rename_volume(self, volume_id, name):
+        self._chvdisk(volume_id, name=name)
+
+    def _chvdisk(self, volume_id, **kwargs):
```

Comment:
no need for `kwargs` if we are not going to use it with more than 1 arguments combination

Reply:
using it [here](https://github.com/IBM/ibm-block-csi-driver/blob/484a0e0ed075ddb8ba9f9e5dc57d51020ef2f731/controller/array_action/array_mediator_svc.py#L1442-L1448) with more arguments..